content/300-guides/300-upgrade-guides/200-upgrading-versions/030-upgrading-to-prisma-4/index.mdx

```diff
@@ -0,0 +1,170 @@
+```
+
+</tab>
+</TabbedContent>
+
+This is a breaking change if you previously had defaults defined for scalar lists at the database level.
+
+### Explicit `@unique` constraints on one-to-one relations
+
+When using one-to-one relations in Prisma 4, you will need to explicitly add the `@unique` attribute on the relation scalar field. For example, for this one-to-one relation between a `User` and a `Profile` model, you will need to add the `@unique` attribute to the `profileId` field:
+
+<TabbedContent tabs={[<FileWithIcon text="Relational databases" icon="database"/>, <FileWithIcon text="MongoDB" icon="database"/>]}>
+<tab>
+
+```prisma
+model User {
+  id        Int      @id @default(autoincrement())
+  profile   Profile? @relation(fields: [profileId], references: [id])
+  profileId Int?     @unique // <-- include this explicitly
+}
+
+model Profile {
+  id   Int   @id @default(autoincrement())
+  user User?
+}
+```
+
+</tab>
+<tab>
+
+```prisma
+model User {
+  id        String   @id @default(auto()) @map("_id") @db.ObjectId
+  profile   Profile? @relation(fields: [profileId], references: [id])
+  profileId String?  @unique @db.ObjectId // <-- include this explicitly
+}
+
+model Profile {
+  id   String @id @default(auto()) @map("_id") @db.ObjectId
+  user User?
+}
+```
+
+</tab>
+</TabbedContent>
+
+### Removing `references` syntax for implicit many-to-many relations
+
+When using [implicit many-to-many relations](/concepts/components/prisma-schema/relations/many-to-many-relations#implicit-many-to-many-relations) in Prisma 4, you will no longer be able to use the `references` argument, which was previously optional. For example, the following relation would now trigger a validation error:
+
+```prisma file=schema.prisma
+model Post {
+  id         Int        @id @default(autoincrement())
+  categories Category[] @relation("my-relation", references: [id]) // <-- validation error
+}
+
+model Category {
+  id    Int    @id @default(autoincrement())
+  posts Post[] @relation("my-relation", references: [id]) // <-- validation error
+}
+```
+
+Instead, you can write:
+
+```prisma file=schema.prisma
+model Post {
+  id         Int        @id @default(autoincrement())
+  categories Category[] @relation("my-relation")
+}
+
+model Category {
+  id    Int    @id @default(autoincrement())
+  posts Post[] @relation("my-relation")
+}
+```
+
+This is because the only valid value for `references` was `id`, so removing this argument makes it clearer what can and cannot be changed.
+
+## Specific upgrade paths
+
+<Subsections />
+
+## Upgrading the <inlinecode>prisma</inlinecode> and <inlinecode>@prisma/client</inlinecode> packages to Prisma 4
+
+To upgrade to Prisma 4 from a `3.x.x` version, you need to update both the `prisma` and `@prisma/client` packages. Both the `prisma` and `@prisma/client` packages install with a caret `^` in their version number. This allows upgrades to new minor versions, but not major versions, to safeguard against breaking changes.
```

Comment:
I do not really understand what the "Both the `prisma` and `@prisma/client` packages install with a caret `^` in their version number." sentence actually means. Which command does this refer to?

Reply:
Yes, but the sentence is the active part that puts them in there with the `^`.

It might refer to running jsut plain `npm install prisma @prisma/client` which then means that just running an install will not install a higher major version, so you have to manually to something to your `package.json` via running an explicit command like the one below - but I am not sure that is what is meant here.